### PR TITLE
Adding documentation for Secrets Rotation

### PR DESCRIPTION
**Pull request by rpothier**
### Desired Outcome

There should be documentation for the Secrets Rotation feature.

### Implemented Changes

Added secrets rotation documentation, some other small doc updates, and updated one of the error messages.
Pulled in an image from [428](secrets-provider-for-k8s/pull/428) that has not landed yet.

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue link: [ONYX-16906](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-16906)

### Definition of Done
[x] Add initial Secrets Rotation documentation.
#### Changelog

- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [X] The changes in this PR do not require tests

#### Documentation

- [X] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [X] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: PUSH_TO_FILE.md [hunk 1]
```diff
@@ -316,14 +316,12 @@ for a description of each environment variable setting:
 | K8s Annotation  | Equivalent<br>Environment Variable | Description, Notes |
 |-----------------------------------------|---------------------|----------------------------------|
 | `conjur.org/authn-identity`         | `CONJUR_AUTHN_LOGIN`  | Required value. Example: `host/conjur/authn-k8s/cluster/apps/inventory-api` |
-| `conjur.org/container-mode`         | `CONTAINER_MODE`      | Allowed values: <ul><li>`init`</li><li>`application`</li><li>`sidecar`</li></ul>Defaults to `init`.<br>Must be set (or default) to `init` or `sidecar`for Push to File mode.|
+| `conjur.org/container-mode`         | `CONTAINER_MODE`      | Allowed values: <ul><li>`init`</li><li>`application`</li></ul>Defaults to `init`.<br>Must be set (or default) to `init`for Push to File mode.|
```
Comment (szh) [inline PUSH_TO_FILE.md:319]: Should we put a note about sidecar mode in parentheses so it doesn't appear to contradict the rotation docs?

File: ROTATION.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+## Troubleshooting
+
+This section describes how to troubleshoot common Secrets Provider for Kubernetes issues.
+
+To enable the debug logs, See [enable-logs](PUSH_TO_FILE.md#enable-logs)
+
+|  Issue      |       Error code   | Resolution  |
+| ----------- | ------------------ | ------ |
+| No change in secret files, no secret files written |CSPFK018I| This is an info message and not an error, the Secrets Provider did not detect a change in secrets. No files will be written. |
```
Comment (szh) [inline ROTATION.md:119]:
```suggestion
| No change in secret files, no secret files written |CSPFK018I| This is an info message and not an error, the Secrets Provider did not detect a change in secrets for the secrets group. The secret file for this group will not be written. Note: there may be changes in other secret groups and those files will be written. |
```

File: ROTATION.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+| Invalid secrets refresh interval annotation |CSPFK050E| There is an error with the interval annotation, check the log message for the exact failure reason. See the [annotation reference](#reference-table-of-configuration-annotations) for more information on setting the annotations.|
+| Invalid secrets refresh configuration |CSPFK051E| Secrets refresh is enabled either by setting `conjur.org/secrets-refresh-enabled` to true or setting a duration for `conjur.org/secrets-refresh-interval` and the mode is not `sidecar`. The mode must be `sidecar` |
```
Comment (szh) [inline ROTATION.md:121]:
```suggestion
| Invalid secrets refresh configuration |CSPFK051E| Secrets refresh is enabled either by setting `conjur.org/secrets-refresh-enabled` to true or setting a duration for `conjur.org/secrets-refresh-interval` and the mode is not `sidecar`. The mode must be `sidecar`. |
```

File: ROTATION.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+## Limitations
+
+This feature is a **Community** level project that is still under development.
+There could be changes to the documentation so please check back often for updates and additions.
+Additional enhancements to this feature include
```
Comment (szh) [inline ROTATION.md:127]:
```suggestion
Future enhancements to this feature include
```

**Review by szh — COMMENTED** [review]
Looks good. I left a few nits

File: ROTATION.md (hunk 2 above)
Comment (rpothier) [reply ROTATION.md:119]: Thanks for the clarification!

**Review by szh — APPROVED** [review]
LGTM! 👍

**Review by alexkalish — CHANGES_REQUESTED** [review]

**Comment by alexkalish** [comment]
@rpothier: Doc overall looks great!  My review is mostly cosmetic.  Thanks.

File: ROTATION.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+## Reference Table of Configuration Annotations
+
+In addition to the [basic Secrets Provider configuration](https://github.com/cyberark/secrets-provider-for-k8s/blob/main/PUSH_TO_FILE.md#reference-table-of-configuration-annotations),
+Below is a list of Annotations that are needed for secrets rotation.
+
+| K8s Annotation  | Description |
+|-----------------------------------------|----------------------------------|
+| `conjur.org/container-mode`         | Allowed values: <ul><li>`init`</li><li>`application`</li><li>`sidecar`</li></ul>Defaults to `init`.<br>Must be set (or default) to `init` or `sidecar`for Push to File mode.|
+| `conjur.org/secrets-refresh-enabled`  |  Defaults to `false` unless `conjur.org/secrets-rotation-interval` is explicitly set. Secrets Provider will exit with error if this is set to `false` and `conjur.org/secrets-rotation-interval` is set. |
```
[the remark this answers is not in the document]
Comment (rpothier) [reply ROTATION.md:108]: The current behaviour is that if `conjur.org/secrets-refresh-enabled` is not set and `conjur.org/secrets-rotation-interval` has a valid duration then refresh will be enabled. Does this behaviour need to change?

[the remark this answers is not in the document]
Comment (alexkalish) [reply ROTATION.md:109]: That behavior is fine.  Mostly, I want the docs to reflect valid values for the annotation.

[the remark this answers is not in the document]
Comment (rpothier) [reply ROTATION.md]: Go it, thanks. Updated the description.

**Review by alexkalish — APPROVED** [review]

**Pull request merged**